Lib/fontmake/__main__.py

```diff
@@ -161,12 +161,23 @@ def main(args=None):
              '%(choices)s. Default: INFO')
     args = vars(parser.parse_args(args))
 
-    project = FontProject(timing=args.pop('timing'),
-                          verbose=args.pop('verbose'))
-
     glyphs_path = args.pop('glyphs_path')
     ufo_paths = args.pop('ufo_paths')
     designspace_path = args.pop('mm_designspace')
+
+    if 'variable' in args['output']:
+        if not (glyphs_path or designspace_path):
+            parser.error(
+                'Glyphs or designspace source required for variable font')
```

Comment:
Thanks Denis. I understand we need at least two fonts, or say masters, to determine axis plane on which the fonts can vary. But what is a designspace? I guess it is the span in which the the VF can be transformed? 

Say, if we have two masters, the designspace is 1D, a segment line.
if we have three masters (not on a same axis), the designspace is a 2D plane (probably a triangle, rectangle, eclipse, etc.).
And [this](https://cdn-images-1.medium.com/max/1600/1*jNGTUj2PX2PXHv6sYUrXCg.png) is a designspace for 3D, a cube.

Is it because we need to avoid intolerant distortions or unreasonable glyphs shapes so that we use designspace to restrict the variation of a font?